apps/tasks/tests.py

```diff
@@ -1,3 +1,211 @@
+import datetime
+
 from django.test import TestCase
+from django.urls import reverse
+from django.utils import timezone
+
+from rest_framework.test import APIClient
+
+from rest_framework_simplejwt.tokens import RefreshToken
+from apps.accounts.models import User
+
+from apps.tasks.models import Task, Comment, TimeLog, Timer
+
+
+class TestTasks(TestCase):
+
+    def setUp(self) -> None:
+        self.client = APIClient()
+        self.user = User.objects.create(first_name='User',
+                                        last_name='For Testing',
+                                        email='string@mail.ogg',
+                                        password='StrongPassword')
+        self.client.force_authenticate(user=self.user, token=f'Bearer {RefreshToken.for_user(self.user)}')
+
+    def test_create_task(self):
+        data = {
+            "title": "string",
+            "description": "string",
+            "assigned_to": self.user.id
+        }
+
+        response = self.client.post(reverse('tasks-list'), data)
+
+        self.assertEqual(201, response.status_code)
+
+    def test_get_tasks(self):
+        Task.objects.create(title='A interesting tile', description='', assigned_to_id=self.user.id)
+
+        response = self.client.get(reverse('tasks-list'))
+
+        self.assertEqual(200, response.status_code)
+
+    def test_get_tasks_by_id(self):
+        Task.objects.create(id=5, title='A interesting tile', description='', assigned_to_id=self.user.id)
+
+        response = self.client.get(reverse('tasks-detail', kwargs={'pk': 5}))
+
+        self.assertEqual(200, response.status_code)
+
+    def test_my_tasks(self):
+        Task.objects.create(title='A interesting tile', description='', assigned_to_id=self.user.id)
+
+        response = self.client.get(reverse('tasks-mytasks'))
+
+        self.assertEqual(200, response.status_code)
+
+    def test_completed_tasks(self):
+        Task.objects.create(title='A complete task', status='completed', description='', assigned_to_id=self.user.id)
+        Task.objects.create(title='Incomplete task', status='to_do', description='', assigned_to_id=self.user.id)
+
+        base_url = reverse('tasks-list')
+        response = self.client.get(base_url + '?status=completed')
+
+        self.assertEqual(200, response.status_code)
+
+    def test_assign_task(self):
+        Task.objects.create(id=3, title='Task to be assigned', description='', assigned_to_id=self.user.id)
+        User.objects.create(id=22, email='test@assign.com', password='DUMMYPASS')
```

Comment:
Asta nu va trai mult, o sa mai adaugi un test inainte de acesta care va crea Task sau User si id'urile deja vor fi ocupate

Reply:
+ password de obicei nu se foloseste in teste, mai aici nu este encoded si chiar daca ai vrea sa testezi login nu va lucra